tests/test_sbr_calculator.py

```diff
@@ -0,0 +1,524 @@
+        electric_heating=True,
+        heating_source=HeatingSource.OTHER,
+        hot_water_source=HotWaterSource.HEAT_BATTERY_OR_ELEC_HOT_WATER_TANK,
+        secondary_heating=False,
+        secondary_hot_water=True,
+        integrated_control_sys=True,
+    )
+    return user_inputs
+
+
+@pytest.fixture(scope="module")
+def test_hp_secondary_heating_hot_water():
+    user_inputs = UserInputs(
+        smart_meter=True,
+        smart_ev_charger=True,
+        charger_power=EVChargerPower.CHARGER_7KW,
+        smart_v2g_enabled=False,
+        home_battery=False,
+        battery_size=BatterySize.NONE,
+        solar_pv=True,
+        pv_inverter_size=SolarInverterSize.LT_4KW,
+        electric_heating=False,
+        heating_source=HeatingSource.OTHER,
+        hot_water_source=HotWaterSource.OTHER,
+        secondary_heating=False,
+        secondary_hot_water=False,
+        integrated_control_sys=True,
+    )
+    return user_inputs
+
+
+@pytest.fixture(scope="module")
+def test_solar_pv():
+    user_inputs = UserInputs(
+        smart_meter=True,
+        smart_ev_charger=False,
+        charger_power=EVChargerPower.NONE,
+        smart_v2g_enabled=False,
+        home_battery=False,
+        battery_size=BatterySize.NONE,
+        solar_pv=True,
+        pv_inverter_size=SolarInverterSize.LT_4KW,
+        electric_heating=False,
+        heating_source=HeatingSource.OTHER,
+        hot_water_source=HotWaterSource.OTHER,
+        secondary_heating=False,
+        secondary_hot_water=False,
+        integrated_control_sys=True,
+    )
+    return user_inputs
+
+
+@pytest.fixture(scope="module")
+def test_smart_meter_only():
+    user_inputs = UserInputs(
+        smart_meter=True,
+        smart_ev_charger=False,
+        charger_power=EVChargerPower.NONE,
+        smart_v2g_enabled=False,
+        home_battery=False,
+        battery_size=BatterySize.NONE,
+        solar_pv=False,
+        pv_inverter_size=SolarInverterSize.NONE,
+        electric_heating=False,
+        heating_source=HeatingSource.OTHER,
+        hot_water_source=HotWaterSource.OTHER,
+        secondary_heating=False,
+        secondary_hot_water=False,
+        integrated_control_sys=False,
+    )
+    return user_inputs
+
+
+@pytest.fixture(scope="module")
+def test_no_smart_meter():
+    user_inputs = UserInputs(
+        smart_meter=False,
+        smart_ev_charger=True,
+        charger_power=EVChargerPower.CHARGER_7KW,
+        smart_v2g_enabled=True,
+        home_battery=True,
+        battery_size=BatterySize.LARGE,
+        solar_pv=True,
+        pv_inverter_size=SolarInverterSize.LT_4KW,
+        electric_heating=True,
+        heating_source=HeatingSource.HEAT_PUMP,
+        hot_water_source=HotWaterSource.HEAT_BATTERY_OR_ELEC_HOT_WATER_TANK,
+        secondary_heating=True,
+        secondary_hot_water=True,
+        integrated_control_sys=False,
+    )
+    return user_inputs
+
+
+@pytest.mark.parametrize(
```

Comment:
Non-blocking: You could also test for negative cases too and check that errors are expected:
https://docs.pytest.org/en/8.1.x/how-to/skipping.html#skip-xfail-with-parametrize